src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -872,8 +872,11 @@ def _strip_prompt(self, token_ids: List[int], prompt_token_id: int, decoder_star
     @staticmethod
     def _convert_to_list(token_ids):
         # convert type to ndarray if necessary
-        if "torch" in str(type(token_ids)) or "tensorflow" in str(type(token_ids)) and hasattr(token_ids, "numpy"):
-            token_ids = token_ids.numpy()
+        if hasattr(token_ids, "numpy"):
+            if "torch" in str(type(token_ids)):
+                token_ids = token_ids.cpu().numpy()
```

Comment:
Following from this - will `token_ids` ever have a grad? In which case, this will also fail on the `cpu` call

Reply:
`token_ids`, the output of `generate`, will not have gradients :) [`generate` is decorated with `@no_grad`](https://github.com/huggingface/transformers/blob/739a63166d30f35b77afda2f968974164b05f33b/src/transformers/generation/utils.py#L1559)